### PR TITLE
refactor(amazonqFeatureDev): improve Error messages

### PR DESCRIPTION
**Pull request by tverney**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description

This PR introduces textual changes in the error and a more granular handling for some specific error scenarios defined by the technical writer.

VS Code version:
https://github.com/aws/aws-toolkit-vscode/pull/5231

## Checklist
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [X] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevController.kt [hunk 1]
```diff
@@ -525,16 +513,31 @@ class FeatureDevController(
...
+                        if (retriesRemaining(session) > 0) {
+                            defaultMessage = message("amazonqFeatureDev.code_generation.error_message")
+                        } else {
+                            defaultMessage = message("amazonqFeatureDev.code_generation.no_retries.error_message")
+                        }
+                    }
+                    else -> {
+                        defaultMessage = null
+                    }
+                }
```
Comment (rli) [inline plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevController.kt:540]:
```suggestion
                val defaultMessage: String? = when (session?.sessionState?.phase) {
                    SessionStatePhase.APPROACH -> {
                        if (isDenyListedError) {
                            message("amazonqFeatureDev.plan_generation.deny_listed_error.failed_generation")
                        } else {
                            message("amazonqFeatureDev.plan_generation.failed_generation")
                        }
                    }
                    SessionStatePhase.CODEGEN -> {
                        if (retriesRemaining(session) > 0) {
                            message("amazonqFeatureDev.code_generation.error_message")
                        } else {
                            message("amazonqFeatureDev.code_generation.no_retries.error_message")
                        }
                    }
                    else -> null
                }
```

**Review by rli — APPROVED** [review]

**Review by chengoramazon — APPROVED** [review]

File: plugins/core/sdk-codegen/codegen-resources/codewhispererruntime/service-2.json
Comment (rli) [inline plugins/core/sdk-codegen/codegen-resources/codewhispererruntime/service-2.json:1]: this diff seems unnecessarily large

**Comment by rli** [comment]
```
> Task :plugin-amazonq:codewhisperer:jetbrains-community:compileTestKotlin FAILED
e: Supertypes of the following classes cannot be resolved. Please make sure you have the required dependencies in the classpath:
    class com.intellij.psi.PsiJavaFile, unresolved supertypes: com.intellij.psi.AbstractBasicJavaFile
Adding -Xextended-compiler-checks argument might provide additional information.

e: file:///codebuild/output/src3109361673/src/github.com/aws/aws-toolkit-jetbrains/plugins/amazonq/codewhisperer/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/codewhisperer/importadder/CodeWhispererJavaImportAdderTest.kt:6:31 Unresolved reference: JavaLanguage
e: file:///codebuild/output/src3109361673/src/github.com/aws/aws-toolkit-jetbrains/plugins/amazonq/codewhisperer/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/codewhisperer/importadder/CodeWhispererJavaImportAdderTest.kt:130:74 Unresolved reference: JavaLanguage
e: file:///codebuild/output/src3109361673/src/github.com/aws/aws-toolkit-jetbrains/plugins/amazonq/codewhisperer/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/codewhisperer/importadder/CodeWhispererJavaImportAdderTest.kt:131:23 Cannot access 'com.intellij.psi.AbstractBasicJavaFile' which is a supertype of 'com.intellij.psi.PsiJavaFile'. Check your module classpath for missing or conflicting dependencies

```

File: plugins/core/sdk-codegen/codegen-resources/codewhispererruntime/service-2.json
Comment (tverney) [reply plugins/core/sdk-codegen/codegen-resources/codewhispererruntime/service-2.json:1]: I just generated from our deployed runtime service and copy/paste. Not sure what is the expected here. I'll try to revert then and manually add the fields that was updated.

**Comment by tverney** [comment]
> ```
> > Task :plugin-amazonq:codewhisperer:jetbrains-community:compileTestKotlin FAILED
> e: Supertypes of the following classes cannot be resolved. Please make sure you have the required dependencies in the classpath:
>     class com.intellij.psi.PsiJavaFile, unresolved supertypes: com.intellij.psi.AbstractBasicJavaFile
> Adding -Xextended-compiler-checks argument might provide additional information.
> 
> e: file:///codebuild/output/src3109361673/src/github.com/aws/aws-toolkit-jetbrains/plugins/amazonq/codewhisperer/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/codewhisperer/importadder/CodeWhispererJavaImportAdderTest.kt:6:31 Unresolved reference: JavaLanguage
> e: file:///codebuild/output/src3109361673/src/github.com/aws/aws-toolkit-jetbrains/plugins/amazonq/codewhisperer/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/codewhisperer/importadder/CodeWhispererJavaImportAdderTest.kt:130:74 Unresolved reference: JavaLanguage
> e: file:///codebuild/output/src3109361673/src/github.com/aws/aws-toolkit-jetbrains/plugins/amazonq/codewhisperer/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/codewhisperer/importadder/CodeWhispererJavaImportAdderTest.kt:131:23 Cannot access 'com.intellij.psi.AbstractBasicJavaFile' which is a supertype of 'com.intellij.psi.PsiJavaFile'. Check your module classpath for missing or conflicting dependencies
> ```

I did not change this file CodeWhispererJavaImportAdderTest or added any extra dependency

File: plugins/core/sdk-codegen/codegen-resources/codewhispererruntime/service-2.json
Comment (rli) [reply plugins/core/sdk-codegen/codegen-resources/codewhispererruntime/service-2.json:1]: try the `-normal.json` in the `c2j` folder

File: plugins/core/sdk-codegen/codegen-resources/codewhispererruntime/service-2.json [hunk 2]
```diff
@@ -1019,12 +1016,6 @@
                 },
                 "codeGenerationStatus": {
                     "shape": "CodeGenerationStatus"
-                },
-                "codeGenerationRemainingIterationCount": {
-                    "shape": "Integer"
-                },
-                "codeGenerationTotalIterationCount": {
-                    "shape": "Integer"
```
Comment (rli) [inline plugins/core/sdk-codegen/codegen-resources/codewhispererruntime/service-2.json:1027]: this was added in https://github.com/aws/aws-toolkit-jetbrains/pull/4662

Comment (tverney) [reply plugins/core/sdk-codegen/codegen-resources/codewhispererruntime/service-2.json]: Yea, I see now. I've got the latest from main, pushed now in the next rev.

File: plugins/core/sdk-codegen/codegen-resources/codewhispererruntime/service-2.json
Comment (tverney) [reply plugins/core/sdk-codegen/codegen-resources/codewhispererruntime/service-2.json:1]: Yea, the diff sounds better. It missed sync with the latest which added some conflicts. It was easier just overwrite and add the difference for the shape I was looking for.

**Comment by rli** [comment]
223 failure is container hitting OOM

**Pull request merged**